lib/browser/api/net.ts

```diff
@@ -481,6 +481,10 @@ export class ClientRequest extends Writable implements Electron.ClientRequest {
   }
 
   _die (err?: Error) {
+    if ((this as any)._writableState.destroyed && err) {
```

Comment:
Can this just be `this.destroyed`? https://github.com/nodejs/node/blob/fe293e914c3b9a65d2024971ebbefcf8a93dc549/lib/_stream_writable.js#L754

Also, I'm not sure I understand why we need to check for `destroyed` here.

Reply:
Node.js assumes that any stream which is ended is no longer capable of emitted events - this is imho a faulty assumption (e.g. in the case of incoming data) but causes this since we are listening for an `error` event emitted _after_ `urlRequest.end()`. Node.js doesn't consider the case of an object that is _acting_ like a stream (our `urlRequest`) for which that behavior is not a reasonable expectation. We also need to check for readable/writable state destruction specifically since that's what Node.js [makes its decision based on](https://github.com/nodejs/node/pull/29197/files#diff-0da8feb832c813299da638cf7fdc65c9R10)

Reply:
ah, so if `destroyed` is true then it will emit the error event itself?